src/rpc/blockchain.cpp

```diff
@@ -960,29 +960,35 @@ static UniValue getblockheader(const JSONRPCRequest& request)
             + HelpExampleRpc("getblockheader", "\"00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09\"")
         );
 
-    LOCK(cs_main);
+    const CBlockIndex* pChainTip;
+    const CBlockIndex* pblockindex;
+    {
+        LOCK(cs_main);
```

Comment:
This can definitely move further down to after validating the parameters; but it definitely needs to be taken before LookupBlockIndex() since it has an AssertLockHeld (but of course see commentary in the mapBlockIndex possibly being it's own lock, in which case we wouldn't need cs_main until it's protecting the chain tip).

Reply:
Yeah, we can tighten this up separately, or with the mapBlockIndex PR (update pending), but this is the minimal change needed to accomplish the unlocking.

Reply:
acceptable